src/reportImport/ui/ReportImportPlugin.php

```diff
@@ -0,0 +1,181 @@
+<?php
+/*
+  Copyright (C) 2014-2016 Siemens AG
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  version 2 as published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License along
+  with this program; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+use Fossology\Lib\Auth\Auth;
+use Fossology\Lib\Dao\FolderDao;
+use Fossology\Lib\Dao\UploadDao;
+use Fossology\Lib\Data\Upload\Upload;
+use Fossology\Lib\Plugin\DefaultPlugin;
+use Symfony\Component\HttpFoundation\Request;
+
+class ReportImportPlugin extends DefaultPlugin
+{
+  const NAME = 'ui_reportImport';
+
+  /** @var UploadDao */
+  private $uploadDao;
+  /** @var FolderDao */
+  private $folderDao;
+
+  function __construct()
+  {
+    parent::__construct(self::NAME, array(
+      self::TITLE => _("Report Import"),
+      self::PERMISSION => Auth::PERM_WRITE,
+      self::REQUIRES_LOGIN => TRUE
+    ));
+    $this->uploadDao = $GLOBALS['container']->get('dao.upload');
+    $this->folderDao = $GLOBALS['container']->get('dao.folder');
+  }
+
+  function preInstall()
+  {
+    $text = _("Import Report");
+    menu_insert("Browse-Pfile::Import&nbsp;Report", 0, self::NAME, $text);
+    menu_insert("Main::Upload::Import&nbsp;Report", 0, self::NAME, $text);
+  }
+
+  protected function handle(Request $request)
+  {
+    $report = intval($request->get('report'));
+    $uploadId = intval(GetArrayVal("upload_pk", $_POST));
+    if (empty($uploadId) ||
+        !array_key_exists('report',$_FILES) ||
+        sizeof($_FILES['report']['name']) != 1)
+    {
+      return $this->showUiToChoose($uploadId);
+    }
+    else
+    {
+
+      $jobMetaData = $this->runImport($uploadId, $_FILES['report'], $request);
```

Comment:
Here the ``$uploadId`` will be always the value of  first upload deom the hidden field.  Adding this line `` $uploadId = intval(GetArrayVal("uploadselect", $_POST));`` in else case will give the value of selected upload.